src/main/java/htsjdk/samtools/util/SnappyLoader.java

```diff
@@ -57,71 +49,47 @@ public SnappyLoader() {
      * @param verbose if true output a small number of debug messages to System.err
      */
     public SnappyLoader(final boolean verbose) {
-        Constructor<InputStream> inputStreamCtor = null;
-        Constructor<OutputStream> outputStreamCtor = null;
-        Class<Error> snappyErrorClass = null;
+        this(verbose, verbose);
+    }
 
+    /**
+     * Constructs a new SnappyLoader which will check to see if snappy is available in the JVM/library path.
+     * @param logFailures  if true output a message to stderr if snappy loading fails
+     * @param logSuccesses if true output a message to stderr if snappy loading succeeds
+     */
+    public SnappyLoader(final boolean logFailures, final boolean logSuccesses) {
         if (java.lang.Boolean.valueOf(System.getProperty("snappy.disable", "false"))) {
             System.err.println("Snappy is disabled via system property.");
+            SnappyAvailable = false;
         }
         else {
+            boolean tmpSnappyAvailable = false;
             try {
-                final Class<InputStream> snappyInputStreamClass = (Class<InputStream>)Class.forName("org.xerial.snappy.SnappyInputStream");
-                final Class<OutputStream> snappyOutputStreamClass = (Class<OutputStream>)Class.forName("org.xerial.snappy.SnappyOutputStream");
-                snappyErrorClass = (Class<Error>)Class.forName("org.xerial.snappy.SnappyError");
-                inputStreamCtor = snappyInputStreamClass.getConstructor(InputStream.class);
-                outputStreamCtor = snappyOutputStreamClass.getConstructor(OutputStream.class, Integer.TYPE);
+                final OutputStream test = new SnappyOutputStream(new ByteArrayOutputStream(1000));
```

Comment:
Previously we tried to be smart about this, but I tend to think the best way to test if snappy compression is available is to actually try and compress something.  It's very hard to test though, since we'd need to be able to fabricate a test that _doesn't_ have access to the native libraries.  My expectation is that the process of opening a stream, writing to it and the closing it should be sufficient.